BlogTemplate/Pages/Edit.cshtml.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using BlogTemplate.Models;
+
+namespace BlogTemplate.Pages
+{
+    public class EditModel : PageModel
+    {
+        private Blog _blog;
+        public Blog Blog { get; set; }
+
+        public EditModel(Blog blog)
+        {
+            _blog = blog;
+        }
+
+        [BindProperty]
+        public Post newPost { get; set; }
+        [BindProperty]
+        public Post oldPost { get; set; }
+        public void OnGet()
+        {
+            InitializePost();
+        }
+
+        private void InitializePost()
+        {
+            string slug = RouteData.Values["slug"].ToString();
+            BlogDataStore dataStore = new BlogDataStore();
+            newPost = oldPost = dataStore.GetPost(slug);
+
+            if(oldPost == null)
+            {
+                RedirectToPage("/Index");
+            }
+        }
+
+        public IActionResult OnPostPublish()
+        {
+            newPost.IsPublic = true;
+            UpdatePost(newPost, oldPost);
```

Comment:
In both of the OnPost* handlers, is oldPost getting bound?  I didn't see any fields in the form for it to bind with...